.github/workflows/windows-ci.yaml

```diff
@@ -0,0 +1,147 @@
+name: Windows-CI
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: windows-latest
+
+    strategy:
+      matrix:
+        arch: ["x86", "x64"]
+        msvc: ["v141"]
+        qt: ["5.12.8"]
+
+    env:
+      CMAKE_GENERATOR: Visual Studio 16 2019
+      CMAKE_GENERATOR_TOOLSET: ${{ matrix.msvc }}
+      vcpkg-triplet: ${{ matrix.arch }}-windows-static-md
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Bootstrap vcpkg
+        id: bootstrap-vcpkg
+        run: |
+          cd C:\vcpkg
+          git pull
+          Write-Host "::set-output name=vcpkg-rev::$(git rev-parse HEAD)"
+          ./bootstrap-vcpkg.bat
+
+      - name: Restore Dependency Cache
+        id: cache-vcpkg
+        uses: actions/cache@v1
+        with:
+          path: C:\vcpkg\installed
+          key: |
+            vcpkg-triplet=${{ env.vcpkg-triplet }} vcpkg-response=${{ hashFiles('vcpkg.txt') }} vcpkg-rev=${{ steps.bootstrap-vcpkg.outputs.vcpkg-rev }}
+          restore-keys: |
+            vcpkg-triplet=${{ env.vcpkg-triplet }} vcpkg-response=${{ hashFiles('vcpkg.txt') }}
+            vcpkg-triplet=${{ env.vcpkg-triplet }}
+
+      - name: Upgrade Dependencies
+        if: steps.cache-vcpkg.outputs.cache-hit == 'true'
+        run: |
+          vcpkg upgrade --no-dry-run --triplet ${{ env.vcpkg-triplet }}
+
+      - name: Build Dependencies
+        run: |
+          vcpkg install `@vcpkg.txt --triplet ${{ env.vcpkg-triplet }}
+          vcpkg list --x-full-desc
+
+      - name: Configure Platform Arch
+        run: |
+          if ("${{ matrix.arch }}" -eq "x86") {
+            $cmake_arch = "Win32"
+            $qt_arch = "win32_msvc2017"
+          } else {
+            $cmake_arch = "${{ matrix.arch }}"
+            $qt_arch = "win64_msvc2017_64"
+          }
+          Write-Host "::set-env name=CMAKE_GENERATOR_PLATFORM::$cmake_arch"
+          Write-Host "::set-env name=QT_ARCH::$qt_arch"
+
+      # Qt5 takes roughly a thousand years to build, so we download it from elsehwere...
+      - name: Restore Qt Cache
+        id: cache-qt
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}\qt
+          key: os=${{ runner.os }} qt=${{ matrix.qt }} arch=${{ env.qt_arch }}
+
+      - name: Install Qt
+        uses: jurplel/install-qt-action@v2
+        with:
+          arch: ${{ env.QT_ARCH }}
+          version: ${{ matrix.qt }}
+          dir: ${{ github.workspace }}\qt
+          cached: ${{ steps.cache-qt.outputs.cache-hit }}
+          aqtversion: ==0.9.*
+
+      # Build KF5SyntaxHighlighting manually due to Qt dependency...
+      - name: Checkout KF5SyntaxHighlighting
+        uses: actions/checkout@v2
+        with:
+          repository: KDE/syntax-highlighting
+          ref: v5.66.0
```

Comment:
Please put the KF5 version in env or matrix.  Also, I would probably download the tarball instead of the repo, since I assume it would be faster than a clone.